internal/pkg/deploy/cloudformation/stack/validate.go

```diff
@@ -29,6 +30,15 @@ var (
 	errInvalidUIDGIDConfig          = errors.New("must specify both UID and GID, or neither")
 	errInvalidEFSConfig             = errors.New("bad EFS configuration: cannot specify both bool and config")
 	errReservedUID                  = errors.New("UID must not be 0")
+	errCircularDependency           = errors.New("bad dependency name: circular dependency present")
+	errInvalidContainer             = errors.New("container dependency does not exist")
+	errInvalidDependsOnStatus       = errors.New("container dependency status must be one of < start | complete | success >")
+	errEssentialContainerStatus     = errors.New("essential container dependencies can only have status 'start'")
+)
+
+var (
+	// Container dependency status options
+	DependsOnStatus = []string{"start", "complete", "success"}
```

Comment:
Is container health check in scheduled job? I believe the blocker was both the LB web service and the scheduled job.

Reply:
good point. i'll adjust the PR to add schedule job as well. maybe ignore this comment until they are all fixed

Reply:
Since you added it to your PR, I added healthy.

Reply:
Does this mean that we need to add some extra validation to stop main containers from depending on `sidecar: healthy` conditions? Since sidecars don't have container health checks there's no way to determine sidecar health.

Reply:
Adjusted methods for finding essential containers and valid status to consider that sidecars can't have healthy condition.